### PR TITLE
take modifications from digitizerMCnew branch

### PR DESCRIPTION
**Pull request by mwinn2**
Improvement and further features in MCH digitisation:
- introduction of MClabels
- better encapsulation of Response and Digitisation
- improvement of parameter handling

### CODE REVIEW

File: Detectors/MUON/MCH/Simulation/include/MCHSimulation/Digit.h [hunk 1]
```diff
@@ -40,9 +40,13 @@ class Digit : public DigitBase
   double getADC() const { return mADC; }
   void setADC(double adc) { mADC = adc; }
 
+  int getLabel() const { return mLabel; }
+  void setLabel(int label) { mLabel = label; }
+
  private:
   int mPadID;
   double mADC;
+  int mLabel;
```
Comment (sawenzel) [inline Detectors/MUON/MCH/Simulation/include/MCHSimulation/Digit.h]: is this actually a label or is it the index where to find the label? (Why is this needed at all?)

Comment (mwinn2) [reply Detectors/MUON/MCH/Simulation/include/MCHSimulation/Digit.h]: @sawenzel : The int represents the index where to find the label. I am not sure, how you the MCtruth matching between tracks and MCtruth hits is later done, but I was thinking that one wants to like to keep track inside the digit to which hit and hence track, the digit belongs to. Otherwise, I don't see how one knows in the digit list which hit starts where exactly. Presumably, I misunderstood what the plan is and the matching is only done based geometrical information later on (not keeping track of this at all basically). 
The same kind of labelindex is also present for the EMCal, which I looked at mostly.

Comment (sawenzel) [reply Detectors/MUON/MCH/Simulation/include/MCHSimulation/Digit.h]: Ok. We were thinking that the associating can be done loosely, by following the convention that the position/index of the digit in the output container is also the index where to lookup the labels. This would keep the digit as close as possible to the actual "raw" data format and we might not need to distinguish between real data and MC data. But of course that depends a bit on detector specificities. Fine for me if you choose to keep the index ... but I would prefer `mLabelIndex` and not just `mLabel` for better readability.

Comment (mwinn2) [reply Detectors/MUON/MCH/Simulation/include/MCHSimulation/Digit.h]: Ok, I renamed it.
I see that it would be attractive not to make the distinction between simulation and real data. However, I have to think about it whether the position is enough to make the connection.

Comment (sawenzel) [reply Detectors/MUON/MCH/Simulation/include/MCHSimulation/Digit.h]: The position will be typically enough (might need some sorting before writing out).

**Comment by sawenzel** [comment]
Can you please fix the formatting problems?

**Comment by mwinn2** [comment]
@sawenzel: i did now the clang format changes that were not yet in place.
